pkg/elasticsearch/client.go

```diff
@@ -179,7 +180,7 @@ func sendEsRequest(cluster, namespace string, payload *EsRequest, client k8sclie
 
 			// Not sure why, but just trying to reuse the request with the old client
 			// resulted in a 400 every time. Doing it this way got a 200 response as expected.
-			sendRequestWithOldClient(cluster, namespace, payload, client)
+			sendRequestWithMTlsClient(cluster, namespace, payload, client)
```

Comment:
this has to do with authn and authz by elasticsearch and EO. in the past we needed to use mTLS but now we can use our SA. so we try to just pass our SA token first, but if that fails (due to the ES version not supporting that method of auth) then we need to fall back to mTLS.
This also helps out our operator in the case where the ES certs are expired -- we can still safely make changes to the cluster in preparation for it to restart.

Reply:
ACK, but doesn't this mean that on normal operations where the cert is not expired we use an SA token. without server certificate verification? Disabling the server cert verification means that a mitm attack is possible and then they get our SA token. Shouldn't the cascade be? First try with SA+server-cert-verification, then SA+insecureTLS, finally mTLS.

Reply:
hmm, i wasn't thinking about the concern that our SA token would be compromised... i'll update to just validate the server cert. (still will be different because we won't provide our client certs)